src/server/auth.c

```diff
@@ -153,7 +154,12 @@ int authorise_server(struct asfd *asfd,
 		logp("unable to get client name\n");
 		goto end;
 	}
-	if(set_string(cconfs[OPT_CNAME], rbuf->buf))
+	cname=strdup(rbuf->buf);
```

Comment:
This should be strdup_w(rbuf->buf, __func__).
Something should check that the result is not NULL:
if(!(cname=strdup_w(rbuf->buf, __func__)))
  goto end;